File: curso/modulo-05/aula-14/exemplo_crypto.py
from Crypto.PublicKey import RSA
from Crypto.Cipher import PKCS1_OAEP
import binascii

parDeChave = RSA.generate(2048)
chavePublica = parDeChave.publickey()

msg = b'desenvolvedorio'

print("========================== CRIPTOGRAFAR =========================", "\n")
encryptor = PKCS1_OAEP.new(chavePublica)
encrypted = encryptor.encrypt(msg)
print("Cipher:", binascii.hexlify(encrypted))

print("========================== DESCRIPTOGRAFAR =========================", "\n")
decryptor = PKCS1_OAEP.new(parDeChave)
decrypted = decryptor.decrypt(encrypted)
print('Clear Text:', decrypted)

File: curso/modulo-05/aula-14/exemplo_crypto.cs
using System.Security.Cryptography;
using System.Text;

static void WriteByteArray(string name, byte[] byteArray)
{
    Console.Write("{0}: {1} bytes, {2}-bit:", name, byteArray.Length, byteArray.Length << 3);
    Console.WriteLine(" -> {0}", BitConverter.ToString(byteArray));
}

var rsa = RSA.Create(2048);

var mensagem = "desenvolvedorio";

Console.WriteLine("========================== CRIPTOGRAFAR =========================");

var cypher = rsa.Encrypt(Encoding.UTF8.GetBytes(mensagem), RSAEncryptionPadding.Pkcs1);
WriteByteArray("Cypher", cypher);

Console.WriteLine("========================== DESCRIPTOGRAFAR =========================");

var clearText = rsa.Decrypt(cypher, RSAEncryptionPadding.Pkcs1);
Console.WriteLine($"Clear text: {Encoding.UTF8.GetString(clearText)}");

